execution_engine/src/execution/executor.rs

```diff
@@ -103,7 +109,15 @@ impl Executor {
                 // These args are passed through here as they are required to construct the new
                 // `Runtime` during the contract's execution (i.e. inside
                 // `Runtime::execute_contract`).
-                runtime.call_contract_with_stack(contract_hash, &entry_point_name, args, stack)
+                match runtime.migrate_contract_and_contract_package(contract_hash) {
```

Comment:
I'd find it more readable:

```rust
runtime.migrate_contract_and_contract_package(contract_hash)?;
runtime.call_contract_with_stack(...)?;
Ok(())
```

Reply:
Tried applying this, but give the return value in this match block the compiler doesn't allow for it